### PR TITLE
Refactor types & Implement loading overlay

### PR DESCRIPTION
**Pull request by Jhuni0123**
- I made `Utils/` directory for our libraries of types and functions which are globally used.
- I made `GlobalMsg` to simply communicate between parents and children. Parents catch the global messages from children and perform update.
- Finally I add loading overlay when submit login form. User can know whether the click submit is really working. It can be used in any http communication in the future.
loading

### CODE REVIEW

File: frontend/src/elm/Utils/Major.elm [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  | DoubleMajor
+
+
+typeToString : MajorType -> String
+typeToString majorType =
+  case majorType of
+    MajorSingle ->
+      "주전공(단일)"
+    MajorMulti ->
+      "주전공(복/부)"
```
Comment (MKRoughDiamond) [inline frontend/src/elm/Utils/Major.elm:23]: Are there differences between "주전공(단일)" and "주전공(복/부)"? Is "주전공(복/부)" just for simplifying the code?
I think it is enough to show users "주전공". Users can see and count the number of tabs.

File: frontend/src/elm/Login/MajorForm/View.elm [hunk 2]
```diff
@@ -6,20 +6,21 @@ import Html.Attributes exposing (id, class, type_, name, value)
...
 view majorForm =
   div [ id "major-info" ]
-    [ div 
+    [ div
         [ ]
         [ header
         , div [ class "table_scroll" ] ( indexedMap majorRow majorForm.majors )
         , newMajor majorForm
         ]
       , div [ id "table-wrapper" ] [ ]
```
Comment (sonowz) [inline frontend/src/elm/Login/MajorForm/View.elm]: Indent is weird here.

File: frontend/src/elm/Login/Msgs.elm [hunk 3]
```diff
@@ -15,3 +15,4 @@ type Msg
   | MajorMsg Login.MajorForm.Msgs.Msg
   | SubmitForm LoginType
   | None
+  | Global GlobalMsg
```
Comment (sonowz) [inline frontend/src/elm/Login/Msgs.elm]: I think `None` should be the last to make code neat.

**Review by sonowz — CHANGES_REQUESTED** [review]
In some files, import order is somewhat messy.
I recommend putting global imports first and local module imports later.

File: frontend/src/elm/Login/MajorForm/View.elm [hunk 4]
```diff
@@ -6,20 +6,20 @@ import Html.Attributes exposing (id, class, type_, name, value)
...
 view : Model -> Html Msg
 view majorForm =
   div [ id "major-info" ]
-    [ div 
-        [ ]
+    [ div []
         [ header
         , div [ class "table_scroll" ] ( indexedMap majorRow majorForm.majors )
         , newMajor majorForm
         ]
```
Comment (sonowz) [inline frontend/src/elm/Login/MajorForm/View.elm]: Between line 16-17, indentation is 4.
I think line 17-20 has tab inside.

**Review by sonowz — CHANGES_REQUESTED** [review]

File: frontend/src/elm/Utils/Major.elm (hunk 1 above)
Comment (Jhuni0123) [reply frontend/src/elm/Utils/Major.elm:23]: Yes there are big differences. Please check http://cse.snu.ac.kr/undergraduate/degree-requirements.
But we can consider changing only "주전공(단일)" to "주전공"

**Review by sonowz — APPROVED** [review]

**Comment by sonowz** [comment]
Good. Now code looks cleaner.

**Pull request merged**